velox/exec/Task.cpp

```diff
@@ -1290,6 +1290,11 @@ ContinueFuture Task::terminate(TaskState terminalState) {
 
   for (auto& [planNodeId, splits] : remainingRemoteSplits) {
     for (auto& split : splits.first) {
+      if (!exchangeClientByPlanNode_[planNodeId]->pool()) {
```

Comment:
Can we handle this logic inside initialize?

Reply:
Hi @xiaoxmeng , upon early termination the exchange operator is not yet created, meaning the client does not yet have the chance to be initialized with the memory pool from exchange operator. But the client will need the pool to create the source, here we just provide the Task's parent pool for the purpose. From my opinion it does not matter because the task is already terminated.

Reply:
I mean the client can have the logic:
```
if (pool_ == nullptr) {
  pool_ = pool
}
```